src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -340,14 +340,15 @@ class ProfileClientActor(requestContext: RequestContext) extends Actor with Fire
       case None => Put
     }
 
-    memberList flatMap { members => pipeline { rawlsRequestMethod(url, members) } }
-
-    // don't leak any sensitive data
-    RequestComplete(NoContent)
-  }.recover {
+    memberList flatMap { members => pipeline { rawlsRequestMethod(url, members) } } map { response =>
```

Comment:
I think something like the alternative you mention will be necessary, because this could take ... say 10 minutes, ballpark?  Also note that we need to consider the "Power Goes Out" situation, so that implies to me something like a PendingWhitelistSyncTable and WhitelistSyncMonitor.